key-visualizer-monitoring-tool.md

```diff
@@ -0,0 +1,160 @@
+---
+title: Key Visualizer
+suammary: Learn the Key Visualizer feature.
+category: how-to
+---
+
+# Key Visualizer
+
+Key Visualizer is a feature for analyzing the usage of TiDB and troubleshooting the traffic hotspot. The feature generates a visual report of the metrics of the TiDB cluster over a period of time. With this visual report, you can quickly and intuitively observe the overall cluster hotspot and traffic distribution.
+
+## Access Key Visualizer
+
+Key Visualizer is one of the TiDB Dashboard components. It is directly integrated to PD instances and does not require a separate deployment. You can access this TiDB Dashboard component on any PD instance via browsers by using the following address:
+
+{{< copyable "" >}}
+
+```
+http://PDAddress:PDPort/dashboard
+```
+
+> **Note:**
+>
+> + If multiple PD instances exist, you can use any PD instance address to access TiDB Dashboard.
+> + By default, `PDPort` is `2379`. If you have modified the corresponding PD parameter, use the corresponding modified PD port.
+> + You need the `root` account of TiDB to log into TiDB Dashboard.
+
+## Interface demonstration
+
+The following image is a demonstration of the Key Visualizer interface:
+
+![Key Visualizer Interface](/media/dashboard/keyvisualizer/overview.png)
+
+From the interface above, you can see the following objects:
+
++ A large heatmap that shows changes of the overall traffic over time.
++ The detailed information of heatmap axises.
++ Information of tables, indexes, and so on (on the left side of the heatmap).
+
+## Basic concepts
+
+This section introduces the basic concepts that relates to Key Visualizer.
+
+### Region
+
+In a TiDB cluster, the stored data is distributed among TiKV instances. Logically, TiKV is a huge and orderly key-value map. The whole key-value space is divided into many segments and each segment consists of a series of adjacent keys. Such segment is called a `Region`.
+
+For detailed introduction of Region, refer to [TiDB Internal (I) - Data Storage](https://pingcap.com/blog/2017-07-11-tidbinternal1/#region).
+
+### Hotspot
+
+The hotspot issue is typical during the usage of TiDB, where high traffic is concentrated on a small range of data. Because consecutive data ranges are often processed on the same TiKV instance, the TiKV instance on which the hotspot occurs becomes the performance bottleneck of the whole application. The hotspot issue commonly occurs in the following scenarios:
```

Comment:
```suggestion
The hotspot issue is typical during the usage of TiDB, where high traffic is concentrated on a small range of data. Because consecutive data ranges are often processed on the same TiKV instance, the TiKV instance on which the hotspot occurs becomes the performance bottleneck of the whole application. The hotspot issue often occurs in the following scenarios:
```